exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/project/ProjectRecordBatch.java

```diff
@@ -120,17 +119,7 @@ protected IterOutcome doWork() {
             // since this is first batch and we already got a NONE, need to set up the schema
             doAlloc(0);
             setValueCount(0);
-
-            // Only need to add the schema for the complex exprs because others should already have
-            // been setup during setupNewSchema
-            for (FieldReference fieldReference : complexFieldReferencesList) {
-              MaterializedField field = MaterializedField.create(fieldReference.getAsNamePart().getName(),
-                      UntypedNullHolder.TYPE);
-              container.add(new UntypedNullVector(field, container.getAllocator()));
-            }
-            container.buildSchema(SelectionVectorMode.NONE);
-            wasNone = true;
-            return IterOutcome.OK_NEW_SCHEMA;
+            return IterOutcome.NONE;
```

Comment:
We haven't built the schema. I think even for the case when incoming operator returned `NONE` for the first batch, we should also build the schema, in the opposite case, we may break some cases of `limit 0` queries.

Reply:
AFAIK there is ```handleNullInput()``` which should be called for this purpose. Here was workaround for adding late complex types to container and rebuilding schema was necessary. One thing I may be missing here is setting ```wasNone = true;```, so I'll add it.

Reply:
Thanks for referring to `handleNullInput()` and explanation.